### ISSUE TITLE
Huge client window on gnome shell

### ISSUE THREAD

**Issue by woprandi** [haiwen/seafile-client#809]
Hi,

I use the seafile client on Fedora 23 (GNOME Shell) with this repo : https://copr.fedorainfracloud.org/coprs/pkerling/seafile/

Since the version 5.1.1 (this issue was not present on 5.0.7), the client is appears "huge" (as if I zoomed)
![image](https://cloud.githubusercontent.com/assets/1449422/16108348/e1a0f074-33a2-11e6-9497-b82fbb7a64c4.png)

With multi monitors, I have not this issue if I drag the window on a external monitor.

**Comment by shoeper** [comment]
I do have the same issue on Fedora 23

**Comment by woprandi** [comment]
After upgrading to Fedora 24. The issue is still there

**Comment by FeuRenard** [comment]
I have the same issue on Fedora 24, GNOME shell. It's annoying, as there is no possibility to resize the window. So the indicators for disk space in the cloud and up-/download cannot be shown.

**Comment by FeuRenard** [comment]
I set up a second system with Fedora 24 and GNOME shell, but the window had a comfortable size.

1. Notebook, 13" display, 1920x1080 => window too large
2. Desktop, 22" display, 1680x1050 => window with comfortable size

I suppose, that this issue correlates with the used screen size. Can you two support this claim, maybe with your screen sizes and resolutions?

**Comment by shoeper** [comment]
14" Lenovo Thinkpad Laptop

**Comment by woprandi** [comment]
@FeuRenard I confirm that.

On external monitors (~22/23", 1920x1080) -> good size
On laptop screen (14", 1920x1080) -> issue

**Comment by gnacho** [comment]
Same problem here :(

**Comment by Loris1123** [comment]
I'm also having this problem on the latest XFCE4. But a friend, using xfce aswell, doesn't have problem.

I'm running Manjaro Linux but switched to xfce a few days ago. I installed/compiled seafile on gnome-shell.

**Comment by wederer** [comment]
I have the same issue on my laptop (1080p) running arch with i3.

**Comment by jhasse** [comment]
On my computer VLC also has this issue, it seems to be something Qt related.

**Comment by jhasse** [comment]
Adding

```
export QT_AUTO_SCREEN_SCALE_FACTOR=0.8
```

to `~/.profile` and rebooting fixed it for me :) Can others confirm?

**Comment by brege** [comment]
jhasse: Good work around!  Can confirm it is working for me, both with Seafile and VLC.

**Comment by gnacho** [comment]
@jhasse Amazing work! Many thanks!! It works like charm, also with Ghostwriter editor!
(I didn't have .profile file, just created with this line) :)

**Comment by fuelflo** [comment]
@jhasse Thx, that works.

How can we include that fix in the seafile client or will this mess up the scaling for others?

**Comment by jhasse** [comment]
I think we'll need to ask the Qt devs about it. I couldn't find a bug report there, should we open one?

**Comment by gnacho** [comment]
Yeah, looks like more QT issue as it's happening also with other programs.

**Comment by flexo3001** [comment]
This problem is due to https://github.com/haiwen/seafile-client/blob/292da5f3cf96cc24d678acafe7dfe4570627e831/src/main.cpp#L70 and ist really annoying. The autoscale feature since QT5.6 sets a 2x scaling on display with >144 Dpi (like my fullhd 12.5" laptop display).

**Comment by gnacho** [comment]
It worked until last update of QT I think (I'm using Antergos).

**Comment by fuelflo** [comment]
Jop, same here....huge window again with client 6.0.7 on Ubuntu Gnome 17.04, gnome 3.2.4.2.

**Comment by FeuRenard** [comment]
I just installed Fedora 26 and now the client window is sized correctly on both of my machines/screens. So this issue doesn't exist any longer for me.

Fedora 26 (Gnome Shell)
Seafile Client 6.0.4

**Comment by woprandi** [comment]
I confirm @FeuRenard, It seems no longer present in Fedora 26

**Comment by flexo3001** [comment]
@FeuRenard Fedora patched QT to use 2x scaling at 192 dpi (like GNOME) instead of 144 dpi. https://bugzilla.redhat.com/show_bug.cgi?id=1381828

**Comment by flexo3001** [comment]
@woprandi it isn't present in Fedora 25 as well. See my previous comment.

**Comment by tobiasmllr** [comment]
I'm on Arch linux and had to add the following lines to ~/.profile
export QT_AUTO_SCREEN_SCALE_FACTOR=1
export QT_SCALE_FACTOR=0.6

The commands are described in http://doc.qt.io/qt-5/highdpi.html
Apparently QT_AUTO_SCREEN_SCALE_FACTOR is a boolean. I guess something about it was changed so that setting it to 0.8 as jhasse suggested does not work anymore.

**Comment by gnacho** [comment]
@tobiasmllr That worked for me. Many thanks!!! :)

**Comment by ludenticus** [comment]
@tobiasmllr Your workaround is OK to Seafile, but it affects other apps as well. For instance QOwnNotes (Qt 5.9) displays tiny Menus and navigation panels. Clementine (Qt 4.8), on the other hand, work as intended.

**Comment by tobiasmllr** [comment]
@ludenticus You can call the QT_* settings for each program separately if you wrap them in a shell script. To test this I removed the QT_ lines from my `./profile` and created the script `seafile_scaled.sh`:
```
#/bin/bash
export QT_AUTO_SCREEN_SCALE_FACTOR=1
export QT_SCALE_FACTOR=0.6
seafile-applet
```
You then have to make the script executable `chmod +x seafile_scaled.sh` and call that instead of `seafile-applet`.

**Comment by fuelflo** [comment]
@tobiasmllr thx, that works like a charm!

**Comment by ludenticus** [comment]
@tobiasmllr Thanks for the workaround. However, as @fuelflo states, now the context menu is quite small. In that case, I prefer the oversized applet, as the «only» problem is that I don't see the bottom of the window

**Comment by flexo3001** [comment]
If you don't use scaling at all, you can disable it by putting `QT_AUTO_SCREEN_SCALE_FACTOR=0` into your /etc/environment, ~/.profile, /etc/profile.d/ etc. It's described here https://blog.qt.io/blog/2016/01/26/high-dpi-support-in-qt-5-6/

**Comment by gegenschall** [comment]
The workarounds described here don't work for me at all. I'm running gnome-shell under wayland.

**Comment by gnacho** [comment]
Problem fixed by itself. I deleted .profile file and apparently everything seems to be working now. I also had some issues with Nexcloud and it's fine as well.
Running Antergos and last versions of the software.

**Comment by gprichter** [comment]
most elegant solution I found was to **modify the .desktop file** for the applet
this will be overwritten when a new packaged version is installed (which hopefully fixes the issue).
to do this:

1. edit the file 
> sudo vim /usr/share/applications/seafile.desktop
2. change the lines, with some explanation why (tendency to forget the reasons behind tweaks :)
```
...
# fixing huge scale problem
# https://github.com/haiwen/seafile-client/issues/809
Exec=env QT_AUTO_SCREEN_SCALE_FACTOR=0 seafile-applet
...
```

**Comment by DerDanilo** [comment]
Why won't the devs add this as option for the client? It's such an easy fix.

**Comment by flexo3001** [comment]
> Why won't the devs add this as option for the client? It's such an easy fix.

Because they don't give a fuck?
